docs/ref/models/querysets.txt

```diff
@@ -2464,6 +2464,55 @@ Class method that returns an instance of :class:`~django.db.models.Manager`
 with a copy of the ``QuerySet``’s methods. See
 :ref:`create-manager-with-queryset-methods` for more details.
 
+
+``explain()``
+~~~~~~~~~~~~~
+
+.. versionadded: 2.1
+
+.. method:: explain(format=None, **options)
+
+Returns the execution plan of the current queryset as a string. This details
+how the database would execute the given query, highlighting any indexes or
+joins used. For example, when using PostgreSQL::
+
+    >>> print(Blog.objects.filter(title="My Blog").explain())
+    Seq Scan on blog  (cost=0.00..35.50 rows=10 width=12)
+      Filter: (title = 'My Blog'::bpchar)
+
+This method is currently supported by all built in database backends with the
+exception of Oracle.
+
+The ``format`` parameter can output the explanation in different formats,
+depending on the database in use. If this parameter is not passed then the
+default database format is used, typically text-based. Currently PostgreSQL
+supports TEXT, JSON, YAML and XML, while MySQL supports TEXT and JSON.
```

Comment:
'and' instead of 'while'? Using 'while' implies MySQL is inferior 😱

Reply:
:D good point, we would not want people to think that.